src/webgpu/util/binary_stream.ts

```diff
@@ -18,9 +18,9 @@ export default class BinaryStream {
     this.view = new DataView(buffer);
   }
 
-  /** buffer() returns the stream's buffer sliced to the 8-byte rounded read or write offset */
+  /** buffer() returns the stream's whole buffer */
   buffer(): ArrayBufferLike {
-    return new Uint8Array(this.view.buffer, align(this.offset, 8)).buffer;
```

Comment:
`new Uint8Array` wraps a view around `this.view.buffer`. `Uint8Array.buffer` returns the underlying `ArrayBuffer`. Therefore this was the same as simply `this.view.buffer`.

Reply:
This was supposed to be `new Uint8Array(this.view.buffer, 0, align(this.offset, 8)).buffer` as in: truncate the array to the written size. I actually noticed this bug yesterday, as all the cached files are 33.6MB (!). Fortunately tar.gz does a good job removing all those 0's.